### PR TITLE
Feature/newtopplaces

### PR DESCRIPTION
**Pull request by giosad**

### CODE REVIEW

File: TopPlaces/TopPlaces.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -59,8 +77,8 @@
 		30730E791C57D4A6001B3668 /* TopPlaces */ = {
 			isa = PBXGroup;
 			children = (
-				30730E9A1C58B92A001B3668 /* Logic */,
-				30730E961C58B8D8001B3668 /* VC */,
+				30DA5FEB1C5E0012007F7A62 /* VC */,
```
Comment (dlahyani) [inline TopPlaces/TopPlaces.xcodeproj/project.pbxproj:80]: We usually name these groups Model and Controllers (and Views if relevant).

File: TopPlaces/TopPlaces/PhotosHistory.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+@import Foundation;
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.h:3]: We use <code>#import</code>. 
You can add common imports to the precompiled header usually named ProjectName-Prefix.pch which is missing in your project for some reason.

Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.h:3]: Add a blank line after and before the import directives.
Fix in the entire project.

File: TopPlaces/TopPlaces/PhotosHistory.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+///
+@interface PhotosHistory : NSObject
+
+/// Add the \c photoInfo to the persistent history log as a first item
+/// It will automatically truncate the storage size to some length, pruning the oldest items
++ (void) addPhotoInfo:(NSDictionary*)photoInfo;
+
+/// Returns the items that were added history recently
++ (NSArray*) historyArray;
+@end
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.h:16]: Add blank line before <code>@end</code>.

File: TopPlaces/TopPlaces/PhotosHistory.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+@import Foundation;
+NS_ASSUME_NONNULL_BEGIN
+
+/// API for accessing persistent history storage.
+///
+@interface PhotosHistory : NSObject
+
+/// Add the \c photoInfo to the persistent history log as a first item
+/// It will automatically truncate the storage size to some length, pruning the oldest items
++ (void) addPhotoInfo:(NSDictionary*)photoInfo;
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.h:12]: No space after return type, i.e. <code>\+ (void)addPhotoInfo:</code>.
Fix across entire project.

File: TopPlaces/TopPlaces/PhotosHistory.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+/// Returns the items that were added history recently
++ (NSArray*) historyArray;
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.h:15]: Same.
I'd suggest using readonly property in that case.

File: TopPlaces/TopPlaces/PhotosHistory.m [hunk 6]
```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+#import "PhotosHistory.h"
+#import "FlickrFetcher.h"
+NS_ASSUME_NONNULL_BEGIN
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.m:6]: Spacing

File: TopPlaces/TopPlaces/PhotosHistory.m [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+#define PHOTOS_HISTORY_PREF_KEY @"history"
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.m:7]: Use static consts instead of defines.

File: TopPlaces/TopPlaces/PhotosHistory.m [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+#define HISTORY_LOG_LENGTH 20
+@implementation PhotosHistory
+
++ (NSArray *) historyArray
+{
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.m:12]: <code>{</code> should be in the same line with function declaration.

File: TopPlaces/TopPlaces/PhotosHistory.m [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];
+  NSArray *photos = [prefs objectForKey:PHOTOS_HISTORY_PREF_KEY];
```
Comment (dlahyani) [inline TopPlaces/TopPlaces/PhotosHistory.m:14]: <code>objectForKey:</code> may return <code>nil</code>, you should mark this method as nullable, or somehow validate that you never  return nil.

File: TopPlaces/TopPlaces/PhotosHistory.h (hunk 5 above)
Comment (giosad) [inline TopPlaces/TopPlaces/PhotosHistory.h:15]: it's a class method - I don't create the instance of PhotosHistory anywhere.

**Pull request closed without merging**